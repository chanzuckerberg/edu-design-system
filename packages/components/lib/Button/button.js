"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const ClickableStyle_1 = __importDefault(require("../ClickableStyle"));
/**
 * Component for making buttons that do not navigate the user to another page.
 *
 * This component is called `Button` because it should be used to make `<button>` elements;
 * however, it can be styled to look like a link.
 *
 * If you need to style an `<a>` element to look like a button, please use the `Link` component.
 * If you need to style a different element or component (e.g. `Link` from `react-router`) to
 * look like a button or link, you can use the `ClickableStyle` component.
 *
 * In terms of the look and feel of the component in the UI, the `Button`, and `Link`, and `ClickableStyle`
 * components are exactly the same.
 */
const Button = (0, react_1.forwardRef)((_a, ref) => {
    var { variant = "flat", color = "brand", disabled = false, type = "button", size = "large" } = _a, rest = __rest(_a, ["variant", "color", "disabled", "type", "size"]);
    return (react_1.default.createElement(ClickableStyle_1.default, Object.assign({}, rest, { as: "button", variant: variant, color: color, disabled: disabled, type: type, size: size, ref: ref })));
});
Button.displayName = "Button";
exports.default = Button;
//# sourceMappingURL=button.js.map