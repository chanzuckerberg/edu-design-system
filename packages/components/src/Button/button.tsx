import Clickable, { ClickableProps } from "../Clickable";
import React, { ReactNode } from "react";

type ButtonHTMLElementProps = React.ButtonHTMLAttributes<HTMLButtonElement>;

export type ButtonProps = ButtonHTMLElementProps & {
  as?: "button" | React.ComponentType<any>;
  /**
   * The button contents or label.
   */
  children: ReactNode;
  color?: ClickableProps<"button">["color"];
  size?: ClickableProps<"button">["size"];
  variant?: ClickableProps<"button">["variant"];
};

function Button({
  as = "button",
  variant = "flat",
  color = "brand",
  disabled = false,
  type = "button",
  size = "medium",
  ...rest
}: ButtonProps) {
  return (
    <Clickable
      {...rest}
      as={as}
      variant={variant}
      color={color}
      disabled={disabled}
      type={type}
      size={size}
    />
  );
}

export default Button;
