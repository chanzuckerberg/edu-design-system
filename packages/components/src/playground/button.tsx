import tw, { styled } from "twin.macro";
import React from "react";
import Typography from "../typography/typography";

type BaseButtonProps = {
  variant?: "primary" | "secondary";
};

// Temporary props for button to test out css options for variants
export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
  BaseButtonProps;

const ButtonComponent = styled.button<BaseButtonProps>(({ variant }) => [
  tw`py-2 px-4 rounded border-none`,
  variant === "secondary"
    ? tw`bg-neutral-500 hover:bg-neutral-600`
    : tw`bg-brand-600 hover:bg-brand-700`,
]);

function Button(props: ButtonProps): JSX.Element {
  const { children, ...rest } = props;

  return (
    <ButtonComponent {...rest}>
      <Typography bold color="white" size="button">
        {children}
      </Typography>
    </ButtonComponent>
  );
}

export default Button;
