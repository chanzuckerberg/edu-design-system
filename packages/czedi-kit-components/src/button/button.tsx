import * as React from "react";

function Button(
  props: React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  >
) {
  return <button {...props}></button>;
}

export default Button;
