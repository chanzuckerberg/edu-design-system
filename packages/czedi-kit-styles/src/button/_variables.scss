$local-variable: 0 0 0 0;