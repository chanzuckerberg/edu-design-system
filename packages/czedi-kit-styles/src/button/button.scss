@import './variables';
@import '@chanzuckerberg/czedi-kit-tokens/dist/scss/variables.scss';



.button {
  color: $color-base-gray-light;
  margin: $local-variable;
  padding: 20px;
}