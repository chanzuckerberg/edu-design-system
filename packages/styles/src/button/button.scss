@import "./_variables.scss";
@import "@chanzuckerberg/czedi-kit-tokens/scss/_variables.scss";

.button {
  color: $color-base-gray-light;
  margin: $local-variable;
  padding: 20px;
  // test rtlcss
  text-align: left;
  float: left;
}
