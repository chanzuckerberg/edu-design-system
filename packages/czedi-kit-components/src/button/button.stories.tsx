import * as React from "react";
import { action } from "@storybook/addon-actions";
import Button from "./button";

export default {
  title: "Button",
  component: Button
};

export const text = () => (
  <Button onClick={action("clicked")}>Hello Button</Button>
);

export const emoji = () => (
  <Button onClick={action("clicked")}>
    <span aria-label="so cool">😀 😎 👍 💯</span>
  </Button>
);
