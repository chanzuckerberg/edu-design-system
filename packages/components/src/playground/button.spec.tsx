import { axe } from "jest-axe";
import { render } from "@testing-library/react";
import { variants } from "./button.stories";

describe("<Button />", () => {
  it("renders the component", () => {
    const { container } = render(variants());
    expect(container.firstChild).toMatchInlineSnapshot(`
      <div>
        <button
          class="button__ButtonComponent-ak17f2-0 cKlymw"
        >
          <p
            class="typography__TypographyComponent-sc-8fjfkf-0 earSjG"
            color="white"
          >
            Hello Button
          </p>
        </button>
        <button
          class="button__ButtonComponent-ak17f2-0 dMLRAo"
        >
          <p
            class="typography__TypographyComponent-sc-8fjfkf-0 earSjG"
            color="white"
          >
            Secondary Button
          </p>
        </button>
      </div>
    `);
  });

  it("passes accessibility checks", async () => {
    const { container } = render(variants());
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
