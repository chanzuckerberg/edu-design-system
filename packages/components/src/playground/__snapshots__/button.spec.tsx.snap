// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> emoji story renders snapshot 1`] = `
.emotion-0 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 0.25rem;
  border-style: none;
  color: var(--eds-color-white);
  font-weight: 700;
  font-size: var(--legacy-size-font-sm);
  line-height: var(--legacy-size-line-height-body);
  background-color: var(--eds-color-brand-600);
}

.emotion-0:hover {
  background-color: var(--eds-color-brand-700);
}

<button
  aria-label="so cool"
  class="emotion-0"
>
  üòÄ üòé üëç üíØ
</button>
`;

exports[`<Button /> primary story renders snapshot 1`] = `
.emotion-0 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 0.25rem;
  border-style: none;
  color: var(--eds-color-white);
  font-weight: 700;
  font-size: var(--legacy-size-font-sm);
  line-height: var(--legacy-size-line-height-body);
  background-color: var(--eds-color-brand-600);
}

.emotion-0:hover {
  background-color: var(--eds-color-brand-700);
}

<button
  class="emotion-0"
>
  Primary Button
</button>
`;

exports[`<Button /> secondary story renders snapshot 1`] = `
.emotion-0 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 0.25rem;
  border-style: none;
  color: var(--eds-color-white);
  font-weight: 700;
  font-size: var(--legacy-size-font-sm);
  line-height: var(--legacy-size-line-height-body);
  background-color: var(--eds-color-neutral-500);
}

.emotion-0:hover {
  background-color: var(--eds-color-neutral-600);
}

<button
  class="emotion-0"
>
  Secondary Button
</button>
`;

exports[`<Button /> text story renders snapshot 1`] = `
.emotion-0 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 0.25rem;
  border-style: none;
  color: var(--eds-color-white);
  font-weight: 700;
  font-size: var(--legacy-size-font-sm);
  line-height: var(--legacy-size-line-height-body);
  background-color: var(--eds-color-brand-600);
}

.emotion-0:hover {
  background-color: var(--eds-color-brand-700);
}

<button
  class="emotion-0"
>
  Hello Button
</button>
`;
